covid19_sfbayarea/data/sonoma.py

```diff
@@ -117,24 +117,83 @@ def transform_cases(cases_tag: element.Tag) -> Dict[str, TimeSeries]:
-    a dictionary of type:
-    {'community': -1, 'from_contact': -1, 'travel': -1, 'unknown': -1}
+    a dictionary. Fields are either the original from data source or are normalized
+    into groups consistent with other datasets, by using `normalize=True` (default).
+    Data model:
+      {'community': -1, 'from_contact': -1, 'travel': -1, 'unknown': -1}
     """
     transmissions = {}
     rows = parse_table(transmission_tag)
+
     # turns the transmission categories on the page into the ones we're using
-    transmission_type_conversion = {'Community': 'community', 'Close Contact': 'from_contact', 'Travel': 'travel', 'Under Investigation': 'unknown'}
+    transmission_type_conversion = {
+        'congregate care': 'congregate_care',
+        'health care': 'health_care',
+        'household': 'household',
+        'large gathering': 'gathering_large',
+        'other': 'other',
+        'small gathering': 'gathering_small',
+        'travel': 'travel',
+        'unknown': 'unknown',
+        'workplace': 'workplace'
+    }
+
     assert_equal_sets(transmission_type_conversion.keys(),
-                      (row['Source'] for row in rows),
+                      (row['Exposure Location'].lower() for row in rows),
                       description='Transmission types')
+
     for row in rows:
-        type = row['Source']
-        number = parse_int(row['Cases'])
+        type = row['Exposure Location'].lower()
+        percent_cases = float(row['All Time'].replace('%', '')) / 100
+        case_count = int(percent_cases * total_cases)
         type = transmission_type_conversion[type]
-        transmissions[type] = number
+        transmissions[type] = case_count
+
+    if normalize:
+        # normalize categories into groups consistent with other datasets
+        # those groups are "from_contact", "travel", "unknown" and "community"
+
+        from_contact_categories = [
+            "congregate_care",
+            "household",
+            "workplace",
+            "gathering_small"
+        ]
+        from_contact = sum(
+            [transmissions.get(category, 0) for category in transmissions.keys()
+             if category in from_contact_categories]
+        )
+        if from_contact == 0:
+            from_contact = -1
+
+        community_categories = [
+            "health_care",
+            "gathering_large",
+            "other"
+        ]
+        community = sum(
+            [transmissions.get(category, 0) for category in transmissions.keys()
+             if category in community_categories]
+        )
+        if community == 0:
+            community = -1
+
+        normalized_transmissions = {}
+        normalized_transmissions["from_contact"] = from_contact
+        normalized_transmissions["community"] = community
+        normalized_transmissions["travel"] = transmissions.get("travel", -1)
+        normalized_transmissions["unknown"] = transmissions.get("unknown", -1)
```

Comment:
Given the above simplifications, this might be nicer and easier to follow as a literal:

```py
        normalized_transmissions = {
            "from_contact": sum(transmissions[category]
                                for category in from_contact_categories),
            "community":    sum(transmissions[category]
                                for category in community_categories),
            # etc.
        }
```